datafusion/physical-expr/src/expressions/binary/kernels.rs

```diff
@@ -125,10 +140,42 @@ pub(crate) fn bitwise_shift_right(left: ArrayRef, right: ArrayRef) -> Result<Arr
             binary_bitwise_array_op!(
                 left,
                 right,
-                |a: i64, b: i64| a.wrapping_shr(b as u32),
+                |a: i64, b: i64| a.wrapping_shr((b as u64).try_into().unwrap()),
                 Int64Array
             )
         }
+        DataType::UInt8 => {
+            binary_bitwise_array_op!(
+                left,
+                right,
+                |a: u8, b: u8| a.wrapping_shr(b as u32),
```

Comment:
TIL that wrapping_shr for `i8` does in fact take a `u32 input 🤷  It looked strange to me but I double checked https://doc.rust-lang.org/std/primitive.i8.html#method.wrapping_shr ✅